lib/runner/cli/cli.js

```diff
@@ -380,14 +380,25 @@ class CliRunner {
     }
 
     for (const env of this.testEnvArray) {
-      if (isMobile(this.testEnvSettings[env].desiredCapabilities)) {
+      const {desiredCapabilities = {}, webdriver = {}} = this.testEnvSettings[env];
+      
+      if (isMobile(desiredCapabilities)) {
         
         if  (Concurrency.isChildProcess()) {
           Logger.info('Disabling parallelism while running tests on mobile platform');
         }
         
         return false;
       }
+
+      if (isSafari(desiredCapabilities) && isLocalhost(webdriver)) {
+        if (Concurrency.isMasterProcess()) {
+          process.argv.push('--serial');
+          Logger.info('Disabling parallelism while running tests on safari browser');
```

Comment:
@beatfactor, Should we add another warning along with this?

Reply:
We should treat `process.argv` as read-only and not change it. `Logger.info` it's not a warning, it will be missed unless you run with `--verbose`. It needs to be a `console.warn()` statement.